### PR TITLE
feat(Table): Add optional icons to tree table rows, add responsive view

### LINKED ISSUE #5598: Table: Tree table - add icon to tree rows
Allow an icon to be included in each row to make the core examples.

### PR DESCRIPTION
**Pull request by nicolethoen**
**What**: Closes #5598 and #5576

 - Adds optional icons to tree table rows - demonstrated in the tree table examples
 - Adds responsive layout to all tree table examples

### CODE REVIEW

**Review by mcarrano — DISMISSED** [review]
This looks great @nicolethoen !

File: packages/react-table/src/components/Table/TreeRowWrapper.tsx [hunk 1]
```diff
@@ -11,15 +12,27 @@ export const TreeRowWrapper: React.FunctionComponent<RowWrapperProps> = ({
...
     <Tr
       aria-level={level}
       aria-posinset={posinset}
       aria-setsize={setsize}
       aria-expanded={!!isExpanded}
       isHidden={isHidden}
-      className={css(className, isExpanded && 'pf-m-expandable', isExpanded && styles.modifiers.expanded)}
+      className={css(
+        className,
+        isExpanded && 'pf-m-expandable',
```
Comment (tlabaj) [inline packages/react-table/src/components/Table/TreeRowWrapper.tsx:32]: @nicolethoen  is this class not in the style sheet?  If not,  @mcoker should this be added?

File: packages/react-table/src/components/TableComposable/Td.tsx [hunk 2]
```diff
@@ -85,10 +86,12 @@ export interface TdProps extends BaseCellProps, Omit<React.HTMLProps<HTMLTableDa
     props?: any;
   };
   treeRow?: {
-    /** */
+    /** Callback when user expands/collapses a row to reveal/hide the row's children */
     onCollapse: OnTreeRowCollapse;
-    /** */
-    onCheckChange: OnCheckChange;
+    /** (optional) Callback when user changes the checkbox on a row */
```
Comment (tlabaj) [inline packages/react-table/src/components/TableComposable/Td.tsx:91]: :+1: Thanks for adding descriptions!

File: packages/react-table/src/components/Table/TreeRowWrapper.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-table/src/components/Table/TreeRowWrapper.tsx:32]: Sometimes classes are just placeholders for the future and those don't appear in the styles object. Not sure if this is the case here.

**Review by kmcfaul — APPROVED** [review]
LGTM!

**Review by mattnolting — DISMISSED** [review]
Looks awesome, thanks @nicolethoen!

File: packages/react-table/src/components/Table/TreeRowWrapper.tsx (hunk 1 above)
Comment (nicolethoen) [reply packages/react-table/src/components/Table/TreeRowWrapper.tsx:32]: yeah - it's a class name in the core example that does not have any styles attached, so they are not in the styles object

Comment (mcoker) [reply packages/react-table/src/components/Table/TreeRowWrapper.tsx:32]: I don't see `pf-m-expandable` referenced on any table row in core. Also that would typically apply to a row if it's expandable, not just in the expanded state. Should be safe to remove.

File: packages/react-table/src/components/Table/utils/decorators/treeRow.tsx [hunk 3]
```diff
@@ -61,6 +66,21 @@ export const treeRow = (onCollapse: OnTreeRowCollapse, onCheckChange?: OnCheckCh
             </span>
           )}
           {text}
+          {!!onToggleRowDetails && (
+            <span className={css(stylesTreeView.tableTreeViewDetailsToggle)}>
+              <Button
+                variant="plain"
+                aria-expanded={isDetailsExpanded}
+                className={css(isDetailsExpanded && styles.modifiers.expanded)}
```
Comment (mcoker) [inline packages/react-table/src/components/Table/utils/decorators/treeRow.tsx:74]: You shouldn't need this. It isn't in the core docs and I don't see any CSS for it.

```suggestion
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks really good! Left a comment about an expanded class you can remove from the tree-view-details-toggle-button.

Also I noticed this bug with the layout with a row that isn't expandable. @mcarrano I'm assuming it's a valid use case to have expandable and non-expandable rows?

**Comment by nicolethoen** [comment]
> Also I noticed this bug with the layout with a row that isn't expandable. @mcarrano I'm assuming it's a valid use case to have expandable and non-expandable rows?

right now, I built the tree table to have all the rows' details be collapsable, or none of them. So if you're finding an issue with only one row now being collapsable, I'm not even sure how to reproduce it, but It was not a use case I built this for.

**Comment by mcoker** [comment]
> right now, I built the tree table to have all the rows' details be collapsable, or none of them. So if you're finding an issue with only one row now being collapsable, I'm not even sure how to reproduce it, but It was not a use case I built this for.

@nicolethoen sorry forgot to include the screenshot! It's the last row.

Screen Shot 2021-04-15 at 2 44 20 PM

**Comment by nicolethoen** [comment]
@mattnolting Do you know if this is just some default padding or spacing that's missing from the `.pf-c-table__tree-view-title-cell` when there is no expand/collapse arrow?

File: packages/react-table/src/components/Table/utils/decorators/treeRow.tsx [hunk 4]
```diff
@@ -61,6 +66,20 @@ export const treeRow = (onCollapse: OnTreeRowCollapse, onCheckChange?: OnCheckCh
...
+          {!!onToggleRowDetails && (
+            <span className={css(stylesTreeView.tableTreeViewDetailsToggle)}>
+              <Button
+                variant="plain"
+                aria-expanded={isDetailsExpanded}
+                aria-label={showDetailsAriaLabel || 'Show row details'}
+                onClick={event => onToggleRowDetails && onToggleRowDetails(event, rowIndex, content, rowData)}
+              >
+                <span className="pf-c-table__details-toggle-icon">
+                  <EllipsisHIcon />
```
Comment (mcoker) [inline packages/react-table/src/components/Table/utils/decorators/treeRow.tsx:78]: does this need `aria-hidden`?

```suggestion
                  <EllipsisHIcon aria-hidden />
```

File: packages/react-table/src/components/Table/examples/Table.md [hunk 5]
```diff
@@ -2094,17 +2105,25 @@ class TreeTable extends React.Component {
     this.buildRows = ([x, ...xs], level, posinset, isHidden = false) => {
       if (x) {
         const isExpanded = this.state.expandedRows.includes(x.repositories);
+        const isDetailsExpanded = this.state.expandedDetailsRows.includes(x.repositories);
         const isChecked = this.state.checkedRows.includes(x.repositories);
+        let icon = <LeafIcon />;
+        if (x.children) {
+          icon = isExpanded ? <FolderOpenIcon /> : <FolderIcon/>;
```
Comment (mcoker) [inline packages/react-table/src/components/Table/examples/Table.md:2112]:
```suggestion
          icon = isExpanded ? <FolderOpenIcon aria-hidden /> : <FolderIcon aria-hidden />;
```

File: packages/react-table/src/components/Table/examples/Table.md [hunk 6]
```diff
@@ -3358,17 +3399,24 @@ class TreeTable extends React.Component {
       // take the first datum from the array (if any)
       if (x) {
         const isExpanded = this.state.expandedRows.includes(x.repositories);
+        const isDetailsExpanded = this.state.expandedDetailsRows.includes(x.repositories);
         const isChecked = this.state.checkedRows.includes(x.repositories);
+        let icon = <LeafIcon />;
+        if (x.children) {
+          icon = isExpanded ? <FolderOpenIcon /> : <FolderIcon/>;
```
Comment (mcoker) [inline packages/react-table/src/components/Table/examples/Table.md:3406]:
```suggestion
          icon = isExpanded ? <FolderOpenIcon aria-hidden /> : <FolderIcon aria-hidden />;
```

**Review by mcoker — DISMISSED** [review]

**Review by tlabaj — APPROVED** [review]

**Comment by kmcfaul** [comment]
@nicolethoen Looks good but needs to to one last rebase to resolve conflicts!

**Review by kmcfaul — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Pull request merged**